com.unity.netcode.gameobjects/Runtime/Simulator/NetworkTypePresets.cs

```diff
@@ -8,17 +17,72 @@ public class NetworkTypePresets
         const string k_DecentMobileDescription = "Suitable for synchronous multiplayer, except that ping (and overall connection quality and stability) may be quite poor.\n\nExpect to handle players dropping all packets in bursts of 1-60s. I.e. Ensure you handle reconnections.";
         const string k_GoodMobileDescription = "In many places, expect this to be 'as good as' or 'better than' home broadband.";
 
-        public static readonly NetworkSimulatorConfiguration None = NetworkSimulatorConfiguration.Create("None", string.Empty, 0, 0, 0, 0, 0);
-        public static readonly NetworkSimulatorConfiguration HomeBroadband = NetworkSimulatorConfiguration.Create("Home Broadband [WIFI, Cable, Console, PC]", k_BroadbandDescription, 2, 2, 1, 0, 0);
-        public static readonly NetworkSimulatorConfiguration Mobile2G = NetworkSimulatorConfiguration.Create("Mobile 2G [CDMA & GSM, '00]", k_PoorMobileDescription, 400, 200, 5, 0, 0);
-        public static readonly NetworkSimulatorConfiguration Mobile2_5G = NetworkSimulatorConfiguration.Create("Mobile 2.5G [GPRS, G, '00]", k_PoorMobileDescription, 200, 100, 5, 0, 0);
-        public static readonly NetworkSimulatorConfiguration Mobile2_75G = NetworkSimulatorConfiguration.Create("Mobile 2.75G [Edge, E, '06]", k_PoorMobileDescription, 200, 100, 5, 0, 0);
-        public static readonly NetworkSimulatorConfiguration Mobile3G = NetworkSimulatorConfiguration.Create("Mobile 3G [WCDMA & UMTS, '03]", k_PoorMobileDescription, 200, 100, 5, 0, 0);
-        public static readonly NetworkSimulatorConfiguration Mobile3_5G = NetworkSimulatorConfiguration.Create("Mobile 3.5G [HSDPA, H, '06]", k_MediumMobileDescription, 75, 50, 5, 0, 0);
-        public static readonly NetworkSimulatorConfiguration Mobile3_75G = NetworkSimulatorConfiguration.Create("Mobile 3.75G [HDSDPA+, H+, '11]", k_DecentMobileDescription, 75, 50, 5, 0, 0);
-        public static readonly NetworkSimulatorConfiguration Mobile4G = NetworkSimulatorConfiguration.Create("Mobile 4G [4G, LTE, '13]", k_DecentMobileDescription, 50, 25, 3, 0, 0);
-        public static readonly NetworkSimulatorConfiguration Mobile4_5G = NetworkSimulatorConfiguration.Create("Mobile 4.5G [4G+, LTE-A, '16]", k_DecentMobileDescription, 50, 25, 3, 0, 0);
-        public static readonly NetworkSimulatorConfiguration Mobile5G = NetworkSimulatorConfiguration.Create("Mobile 5G ['20]", k_GoodMobileDescription, 1, 10, 1, 0, 0);
+        public static readonly NetworkSimulatorConfiguration None
+#if UNITY_MP_TOOLS_NETSIM_ENABLED
```

Comment:
Seems like we could ifdef the whole class? Things that refer to this will probably not exist, but that might mean we make these internal? Not too sure, but we'll probably have to chat about what this implies, because user code that references presets will just break if we set them to null.

Reply:
If we ifdef the whole class, it prevents building since user code may be referencing those.